File: ics.generator/src/main/java/edu/jimei/ics/generator/IcsGeneratorFactory.java
package edu.jimei.ics.generator;

import edu.jimei.ics.generator.impl.IcsGeneratorImpl;

import java.time.LocalDate;

public class IcsGeneratorFactory {
    public static IcsGenerator createIcsGenerator(LocalDate date) {
        return new IcsGeneratorImpl(date);
    }
}

File: ics.generator/src/main/java/ics/IcsGeneratorFactory.kt
package ics

import ics.impl.IcsGeneratorImpl
import ics.impl.IcsGeneratorImplKt
import java.time.LocalDate

object IcsGeneratorFactory {
    fun createIcsGenerator(date: LocalDate?): IcsGenerator {
        return createIcsGenerator_old(date)
    }

    fun createIcsGenerator_new(date: LocalDate?): IcsGenerator {
        return IcsGeneratorImplKt(date!!)
    }

    fun createIcsGenerator_old(date: LocalDate?): IcsGenerator {
        return object : IcsGeneratorImpl(date) {
        }
    }
}
